libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventStream.kt

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.navigation.core.replay.history
+
+import com.google.gson.Gson
+import com.google.gson.JsonArray
+import com.google.gson.JsonElement
+import com.google.gson.JsonNull
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import com.google.gson.internal.LazilyParsedNumber
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonToken
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import java.io.Closeable
+
+/**
+ * Lower memory option for reading replay events from a history file.
+ */
+class ReplayEventStream(
+    private val jsonReader: JsonReader
```

Comment:
I do have an issue with this requiring Json

an alternative considered, is to make this class an interface. and then move this implementation to "examples". that way we have 1 less dependency on Gson in navigation-core